### PR TITLE
fix(eslint-eqeqeq): updated the `eqeqeq` rule to match the core repo

### LINKED ISSUE #1433: Using gts instead of eslint
I've been having some issues with eslint's configuration as it is configured as part of the gts module. Is there a specific reason for using gts instead of eslint, prettier, etc just like in the core repo?

### PR DESCRIPTION
**Pull request by samimusallam**
Fixes #1433 

## Short description of the changes

- Updated the `eqeqeq` rule in the eslint config to match the core repo.

### CODE REVIEW

File: plugins/node/instrumentation-amqplib/src/amqplib.ts [hunk 1]
```diff
@@ -250,7 +250,7 @@ export class AmqplibInstrumentation extends InstrumentationBase {
         url,
         socketOptions,
         function (this: unknown, err, conn: Connection) {
-          if (err == null) {
+          if (err === null) {
```
[the remark this answers is not in the document]
Comment (samimusallam) [reply plugins/node/instrumentation-amqplib/src/amqplib.ts:253]: Yes, there are a couple of instances similar to this. I'm updating the code to `err === null || err === undefined` for this, and the other similar checks.

[the remark this answers is not in the document]
Comment (Flarna) [reply plugins/node/instrumentation-amqplib/src/amqplib.ts:253]: In core repo we use `!= null`/`== null` and I would vote for doing the same in this repo.

[here](https://github.com/open-telemetry/opentelemetry-js/blob/32ae641ad13be8786cce9ffc3942efa385c0ab33/packages/opentelemetry-sdk-trace-base/src/Span.ts#L115) is one random picked location

[the remark this answers is not in the document]
Comment (samimusallam) [reply plugins/node/instrumentation-amqplib/src/amqplib.ts:253]: Actually, I wasn't aware that the `smart` option allows `==` for null values. will revert these changes :)

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
               return moduleExports;
             },
             (moduleExports?: LambdaModule) => {
-              if (moduleExports == undefined) return;
+              if (moduleExports === undefined) return;
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts:135]:
```suggestion
              if (moduleExports == null) return;
```
To keep behavior as before

**Review by Flarna — APPROVED** [review]

**Pull request merged**